sys/linux/dev_mali.txt

```diff
@@ -331,6 +335,17 @@ kbase_ioctl_context_priority_check {
 	priority	flags[queue_group_priority, int8]
 }
 
+prfcnt_control_cmd {
+	cmd		flags[control_cmd, int8]
+	padding		array[const[0, int16], 3]
+	user_data	ptr64[in, array[int64]]
```

Comment:
Do we have insight into the size of this array and how it's being used?

Reply:
```
/** struct prfcnt_control_cmd - Control command
 * @cmd:       Control command for the session.
 * @pad:       Padding bytes.
 * @user_data: Pointer to user data, which will be returned as part of
 *             sample metadata. It only affects a single sample if used
 *             with CMD_SAMPLE_SYNC or CMD_SAMPLE_ASYNC. It affects all
 *             samples between CMD_START and CMD_STOP if used with the
 *             periodic sampling.
 */
struct prfcnt_control_cmd {
	__u16 cmd;
	__u16 pad[3];
	__u64 user_data;
};
```

As seen in the definition of prfcnt_control_cmd, the user_data field is described as a "pointer to user data." However, this description alone is not sufficient to clearly determine the state of the array. I confirmed that user_data is used in functions such as kbasep_kinstr_prfcnt_client_output_sample to store certain data, but I was unable to gather additional information to definitively conclude its exact state.

I updated the definition to ptr64[out, array[int64]] (changing it from the original ptr64[in, array[int64]]) as I couldn’t think of a more appropriate way to define it. If this definition needs to be revised, I would greatly appreciate your suggestions on how it can be improved. Thank you so much for your valuable advice!

Reply:
I see, thanks. I think we're good for now.